src/main/java/com/atlassian/db/replica/internal/SqlQuery.java

```diff
@@ -1,16 +1,43 @@
 package com.atlassian.db.replica.internal;
 
-public interface SqlQuery {
-    static SqlQuery create(String sql, boolean compatibleWithPreviousVersion){
-        if(compatibleWithPreviousVersion){
-            return new SqlQueryOld(sql);
-        }else{
-            return new SqlQueryNew(sql);
+public final class SqlQuery {
```

Comment:
We can do this in `internal`.
But in API, turning an `interface` to a `class` (or vice versa) would break binary compatibility.

Reply:
That's why we don't expose all the classes as API 😉 .